### PR TITLE
Update android zoom sdk for Aug 2020

### PR DESCRIPTION
**Pull request by Yatanvesh**
The current zoom android sdk support was dropped in July 2020, as a result zoom meetings cannot be joined or hosted.
I've updated the zoom sdk implementation, as well as made other small changes to get the library to run and compile.

This is a semi tested version, I've tested Armeabi and armv64 release and debug builds on my 2 android devices.
Join and create meeting successfully tested.

Needless to say, I didn't update IOS configuration in this PR.

### CODE REVIEW

File: android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java [hunk 1]
```diff
@@ -205,6 +205,10 @@ public void onZoomSDKInitializeResult(int errorCode, int internalErrorCode) {
     }
   }
 
+  @Override
+  public void onZoomAuthIdentityExpired(){
+  }
+
```
Comment (mieszko4) [inline android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java:211]: why is this needed?

File: android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java [hunk 2]
```diff
@@ -110,7 +110,7 @@ public void startMeeting(
       params.userId = userId;
       params.userType = userType;
       params.zoomAccessToken = zoomAccessToken;
-      params.zoomToken = zoomToken;
+//       params.zoomToken = zoomToken;
```
Comment (mieszko4) [inline android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java:113]: could you comment on this?

File: README.md [hunk 3]
```diff
@@ -46,6 +46,8 @@ Note: In `android/app/build.gradle` I tried to set up `compile project(':react-n
 and it compiled well, but the app then crashes after running with initialize/meeting listener.
 So the above solution seems to be the best for now.
 
+Note: If you're using proguard copy the contents of this module's android/proguard.cfg and append them to your project's android/app/proguard-rules.pro otherwise the release build will crash
```
Comment (mieszko4) [inline README.md:49]: could you comment on from where this `android/proguard.cfg` comes from?

File: android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java (hunk 1 above)
Comment (Yatanvesh) [reply android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java:211]: Version of lib files :v5.0.24437.0708
onZoomAuthIdentityExpired is a required abstract method now, without it it simply won't build.

Basically what i did here was took your library, replace the lib files with new ones downloaded from zoom's repo (zoom-sdk-android-master) and then proceeded to make all the changes so that it'll build.

File: android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java (hunk 2 above)
Comment (Yatanvesh) [reply android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java:113]: The commented line gives an error in build. I didn't look deeply but from initial look it seems that the parameter has been removed. The API docs specify that ZAK token is enough to start a meeting, and also zoomToken was previously not required to begin with.
I have tested the build after commenting it, yes I didn't update the readme and method params yet. Will you do that or I should take a look into that?

File: README.md (hunk 3 above)
Comment (Yatanvesh) [reply README.md:49]: This file will be ignored in the build process, I've added it here so that people who use proguard can use its contents. I think it'll be better to move it to a separate readme file, your call.
This file is taken from https://github.com/zoom/zoom-sdk-android root directory and I can confirm that without it the proguard enabled release will crash, and although i wish i could shorten this file by removing unnecessary rules, I didn't have the time when making this PR. 
Maybe I'll try this in a future PR.

File: android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java (hunk 2 above)
Comment (mieszko4) [reply android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java:113]: got it, thnx. A lot has changed in zoom api since I last time updated the lib.
if you feel like updating the readme that would be great. I can probably take a look on that but later.

Btw regarding macos I am still waiting for zoom to support xcode 11 before proceeding.
But it is taking them so long..
image

**Comment by Yatanvesh** [comment]
All the changes I've made are inspired by the intent to get the build to compile.
The targetSdk and compileSdk versions were also upgraded and support libraries were added just for that

File: android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java (hunk 2 above)
Comment (Yatanvesh) [reply android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java:113]: Understood lol, feel free to close this PR and use its changes when xcode 11 support comes out and you're able to test both platforms. Ill be happy to pitch in then and test the builds too.
